crates/api_common/src/community.rs

```diff
@@ -39,10 +41,10 @@ pub struct CommunityResponse {
   pub community_view: CommunityView,
 }
 
-#[derive(Serialize, Deserialize, Debug)]
+#[derive(Serialize, Deserialize, Debug, Default)]
 pub struct ListCommunities {
-  pub type_: Option<String>,
-  pub sort: Option<String>,
+  pub type_: Option<ListingType>,
+  pub sort: Option<SortType>,
```

Comment:
This is an API change, and would require the front end clients to use these enums. I'm not against it, just know that its a breaking change

Reply:
Also, just to be sure, remove the from_opt_str_to_opt_enum function, that way we can get rid of all of these.

Reply:
No, clients work just as before, because serde parses the strings into the correct enum variant. I already tested locally with lemmy-ui, and different sort/listing options worked without any problems.

Reply:
from_opt_str_to_opt_enum function is already removed.

Reply:
That's kind of wild, because every language does enums differently, and a lot don't even have the concept of string enums. In typescript you have to manually define them as string enums, not the standard integer ones. Okay then, hopefully it doesn't break clients, but we'll know when we cut the RC's.

Also I would still want to change lemmy-js-client to use the enums rather than strings as they do now. I'll open up an issue for that.